include/StrokeEngine.h

```diff
@@ -0,0 +1,175 @@
+/**
+ *   StrokeEngine
+ *   A library to create a variety of stroking motions with a stepper or servo motor on an ESP32.
+ *   https://github.com/theelims/StrokeEngine 
+ *
+ * Copyright (C) 2022 theelims <elims@gmx.net>
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license.  See the LICENSE file for details.
+ */
+
+#pragma once
+#ifndef STROKE_ENGINE_H
+#define STROKE_ENGINE_H
+
+#include <pattern.h>
+#include <motor.hpp>
+
+// Debug Levels
+//#define DEBUG_TALKATIVE             // Show debug messages from the StrokeEngine on Serial
+//#define DEBUG_STROKE                // Show debug messaged for each individual stroke on Serial
+#define DEBUG_CLIPPING              // Show debug messages when motions violating the machine 
+                                    // physics are commanded
+
+enum class StrokeParameter {
+  // RATE & SPEED are mutually exclusive. Only one can be specified at a time!
+  // RATE - Range 0.5 to 6000 Strokes / Min
+  // Can allow better control typically than just SPEED, as other machines use
+  RATE,
+
+  // SPEED - Range 1 to 5000 mm / Sec
+  // Standard Speed Control
+  SPEED,
```

Comment:
I think speed is generally not compatible with how my patterns work. Rate should be the much better value. IMHO speed could be left out.